factfeel_client.py

```diff
@@ -125,11 +125,59 @@ def fact_feel_modify_lights(self, prediction):
 
 class SpeechToText:
     
-    def __init__(self):
+    def __init__(self, device = 2, init = False, debug = False):
+        
+        self._debug = debug
+        
+        if debug:
+            self._device_listings()
+        self._device = device
+        
+        if init:
+            self.initialize()
+        else:
+            self.initialized = False
+
+    
+    @property
+    def device(self):
+        return self._device
+    
+    @device.setter
+    def device(self, device):
+        self._device = device
+    
+    def _device_listings(self):
+        import pyaudio
+        audio = pyaudio.PyAudio()
+        
+        num_devices = audio.get_host_api_info_by_index(0).get("deviceCount",0)
+        device_count = 0
+        self.devices = {}
+        for i in range(0, num_devices):
```

Comment:
Say the number of devices detected is non-zero, but the index passed in from the UI layer does not match a valid known device. What should happen in this case? Error, or a default condition where you use a known device instead? I'm leaning towards error as you probably don't want a mic you aren't expecting to be hot

Reply:
I like the error handling approach, I've added two error conditions it is checking for now. The first is checking if this is a valid index given the list of available devices. It will list out available devices within the error. A side note, I can't seem to catch the OSError that gets thrown, so I'm leaving it in for now, my error message still comes out and is trying to be rather explanatory about the reason and fix:
![First Error Handling_Invalid Device](https://user-images.githubusercontent.com/6473640/171045588-3bbef679-262b-4870-8502-1c3922bfcf02.PNG)